src/components/tl/ucp/tl_ucp_coll.h

```diff
@@ -188,13 +188,15 @@ ucc_tl_ucp_init_task(ucc_base_coll_args_t *coll_args, ucc_base_team_t *team)
     ucc_tl_ucp_task_t *task    = ucc_tl_ucp_get_task(tl_team);
 
     ucc_coll_task_init(&task->super, coll_args, team);
-    if (coll_args->mask & UCC_COLL_ARGS_FIELD_TAG) {
+    if (UCC_COLL_ARGS_ACTIVE_SET(&coll_args->args)) {
+        task->subset.map    = coll_args->args.active_set.map;
+        task->subset.myrank = (ucc_rank_t)coll_args->args.active_set.my_ep;
+        /* ActiveSet of size 2 can use a dedicated bit in the tag space.
+           Otherwise we need to generate a uniq ID for an Active Set */
+        ucc_assert(coll_args->args.active_set.map.ep_num == 2);
+        task->tagged.tag = UCC_TL_UCP_MAX_TAG;
```

Comment:
I don't think you want to always set this to `UCC_TL_UCP_MAX_TAG`. Shouldn't this just set the high order bit on its own? If the user is providing their own tag, we want to honor that. So the check for an active set would be orthogonal to the check for `UCC_COLL_ARGS_FIELD_TAG` (i.e., not in the same if/else block).